crates/red_knot_python_semantic/mdtest.py

```diff
@@ -0,0 +1,200 @@
+"""A runner for Markdown-based tests for Red Knot"""
+# /// script
+# requires-python = ">=3.11"
+# dependencies = [
+#     "rich",
+#     "watchfiles",
+# ]
+# ///
```

Comment:
Should we add a `tool.uv.exclude-newer` value in this inline config, to ensure new versions of these dependencies don't unexpectedly break the script? I doubt that they will -- `rich` and `watchfiles` are both very widely used and fairly dependable... but there's still a chance that one of them might do a breaking major release at some point?

Reply:
We could also use the brand new locking for scripts: https://docs.astral.sh/uv/guides/scripts/#locking-dependencies.